queue.go

```diff
@@ -20,8 +23,19 @@ var (
 
 	consumerWid  uint64 = 0
 	publisherWid uint64 = 0
+	once         sync.Once
 )
 
+type config struct {
```

Comment:
please rename it to `URL`

also you can move that assembly logic to that type like this
```golang
func (u *Url) String() {
    return fmt.Sprintf("amqp://%s:%s@%s:%s/%s", u.UserName, u.Password, u.HostName, u.Port, u.Vhost),
}
```
so it is easier to track the code